### ISSUE TITLE
Support 64-bit datetime values

### ISSUE THREAD

**Issue by aquilax** [GothenburgBitFactory/taskwarrior#2017]
#### To report a bug...

* What command(s) did you run?
```bash
$ task add due:2045-12-17 project:personal Test
'-1897889296' is not a valid date in the '' format.
```
* What did you expect to happen?
Task to be added
* What actually happened?
`'-1897889296' is not a valid date in the '' format.`
* Paste the output of the ```task diag``` command.
```bash
task 2.5.1
   Platform: Linux

Compiler
    Version: 5.4.0 20160609
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      CMake: 3.5.1
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.4.10
 Build type: None

Configuration
       File: /home/aquilax/.taskrc (found), 1839 bytes, mode 100664
       Data: /home/aquilax/ledger/task (found), dir, mode 40775
    Locking: Enabled
         GC: Enabled
     Server: taskwarrior.inthe.am:53589
         CA: /home/aquilax/ledger/task_sync/ca.cert.pem, readable, 2405 bytes
      Trust: ignore hostname
Certificate: /home/aquilax/ledger/task_sync/private.certificate.pem, readable, 2462 bytes
        Key: /home/aquilax/ledger/task_sync/private.key.pem, readable, 3664 bytes
    Ciphers: NORMAL
      Creds: inthe_am/aquilax/************************************

Hooks
     System: Enabled
   Location: /home/aquilax/ledger/task/hooks
             (-none-)

Tests
      $TERM: xterm (322x70)
       Dups: Scanned 510 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 510 tasks for broken references:
             No broken references found

```

**Comment by pbeckingham** [comment]
This is the unfortunate side effect of using `time_t` structures.  This will be addressed, once we have a good suggested solution.  Thanks for the reminder.

**Comment by tomato42** [comment]
but `time_t` is 64 bit on 64 bit systems... is it overridden somewhere?

(yes, there are [problems](https://lwn.net/Articles/643234/) on 32 bit systems that can't really be workarounded easily)

**Comment by tbabej** [comment]
This is correct, and since kernel version 5.6 even 32 bit systems should be able to handle 64 bit time_t. Currently our code does translate (at least on my machine) the time_t type to 64 bit representation, as evidenced by:

```
$ task diag | grep time_t
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
```

I added a test in 1a52c593568c0598321d90d34510d25556f7f8fe to see if this is preserved across tested platforms.

**Comment by tbabej** [comment]
Implemented in 2.6.0. Taskwarrior now supports Datetime values up until year 9999.
